librz/core/cmd/cmd_print.c

```diff
@@ -7198,3 +7198,674 @@ CMD_PRINT_BYTE_ARRAY_HANDLER_NORMAL(rz_cmd_print_byte_array_swift_handler, RZ_LA
+		rz_core_block_size(core, old_blocksize);
+		rz_core_block_read(core);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_disassembly_all_possible_opcodes_treeview_handler(RzCore *core, int argc, const char **argv) {
+	bool color = rz_config_get_i(core->config, "scr.color") > 0;
+	RzAsmOp asm_op = { 0 };
+	const ut32 n_bytes = 28; // uses 56 chars
+	ut32 old_blocksize = core->blocksize;
+	const char *pal_reg = core->cons->context->pal.reg;
+	const char *pal_num = core->cons->context->pal.num;
+
+	if (old_blocksize < n_bytes) {
+		rz_core_block_size(core, 256);
+		rz_core_block_read(core);
+	}
+
+	rz_cons_break_push(NULL, NULL);
+	for (ut32 position = 0; position < n_bytes && !rz_cons_is_breaked(); position++) {
+		ut64 offset = core->offset + position;
+		rz_asm_set_pc(core->rasm, offset);
+		ut8 *buffer = core->block + position;
+		ut32 length = RZ_MAX(n_bytes - position, core->blocksize - position);
+		int op_size = rz_asm_disassemble(core->rasm, &asm_op, buffer, length);
+		if (op_size < 1) {
+			continue;
+		}
+		op_size = RZ_MAX(op_size, 1);
+		char *op_hex = rz_hex_bin2strdup(buffer, op_size);
+		char *assembly = strdup(op_size > 0 ? rz_asm_op_get_asm(&asm_op) : "illegal");
+		char *colored = NULL;
+
+		if (color) {
+			RzAnalysisOp aop = { 0 };
+			rz_analysis_op(core->analysis, &aop, offset, buffer, length, RZ_ANALYSIS_OP_MASK_ALL);
+			char *tmp = rz_print_colorize_opcode(core->print, assembly, pal_reg, pal_num, false, 0);
+			colored = rz_str_newf("%s%s" Color_RESET, rz_print_color_op_type(core->print, aop.type), tmp);
+			free(tmp);
+		}
+
+		int padding = position * 2;
+		int space = 60 - padding;
+
+		if ((position + op_size) >= 30) {
+			ut32 last = (30 - position) * 2;
+			op_hex[last - 1] = '.';
+			op_hex[last] = 0;
+		}
+
+		rz_cons_printf("0x%08" PFMT64x " %*s%*s %s\n", offset, padding, "", -space, op_hex, colored ? colored : assembly);
+		free(op_hex);
+		free(assembly);
+		free(colored);
+	}
+	rz_cons_break_pop();
+
+	if (old_blocksize < n_bytes) {
+		rz_core_block_size(core, old_blocksize);
+		rz_core_block_read(core);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_disassembly_basic_block_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	RzAnalysisBlock *b = rz_analysis_find_most_relevant_block_in(core->analysis, core->offset);
+	core->num->value = 0;
+	if (!b) {
+		RZ_LOG_ERROR("Cannot find function at 0x%08" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	ut8 *block = malloc(b->size + 1);
+	if (!block) {
+		RZ_LOG_ERROR("Cannot allocate buffer\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_io_read_at(core->io, b->addr, block, b->size);
+
+	rz_cmd_state_output_array_start(state);
+	switch (state->mode) {
+	case RZ_OUTPUT_MODE_STANDARD:
+		rz_core_print_disasm(core->print, core, b->addr, block, b->size, 9999, 0, 2, false, NULL, NULL);
+		break;
+	case RZ_OUTPUT_MODE_JSON:
+		rz_core_print_disasm_json(core, b->addr, block, b->size, 0, state->d.pj);
```

Comment:
The same concern applies here - shouldn't we pass `state` instead and merge these two functions?